javaharness/java/arcs/android/AndroidArcsEnvironment.java

```diff
@@ -98,6 +110,41 @@ void init(Context context) {
     // needed to allow WebWorkers to work in FileURLs.
     arcsSettings.setAllowUniversalAccessFromFileURLs(true);
 
+    // As trampolines to map https protocol to file protocol.
+    // E.g. https://appassets.androidplatform.net/assets/foo is mapped to
+    // file:///android_asset/foo
+    final WebViewAssetLoader assetLoader = new WebViewAssetLoader.Builder()
+        .addPathHandler("/assets/", new AssetsPathHandler(context))
+        .addPathHandler("/res/", new ResourcesPathHandler(context))
+        .build();
+
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+        // For the renderer main thread to intercept the urls.
+        return assetLoader.shouldInterceptRequest(request.getUrl());
+      }
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+        Trace.beginAsyncSection("AndroidArcsEnvironment::init::loadUrl", 0);
+      }
+
+      @Override
+      public void onPageFinished(WebView view, String url) {
+        Trace.endAsyncSection("AndroidArcsEnvironment::init::loadUrl", 0);
+      }
+    });
+
+    ServiceWorkerController swController = ServiceWorkerController.getInstance();
+    swController.setServiceWorkerClient(new ServiceWorkerClient() {
```

Comment:
Nit: you may as well chain these calls and not use the temporary variable:

```
ServiceWorkerController.getInstance()
    .setServiceWorkerClient(...);
```

Reply:
Done.